### PR TITLE
MDEXP-69: Update a jobExecution entry in DB

### PR DESCRIPTION
**Pull request by IlliaDaliek**
## Purpose
Update status of job execution in database in pair with file definition export process.

### CODE REVIEW

File: src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java [hunk 1]
```diff
@@ -82,24 +88,63 @@ protected void initBlocking(JsonObject request, JsonObject params) {
...
+  /**
+   * Update jobExecution status with IN-PROGRESS && update exported files && update completed date
+   *
+   * @param jobExecution         job execution
+   * @param fileExportDefinition definition of the file to export
+   * @param tenantId             tenant id
+   */
+  private void updateJobExecutionStatusAndExportedFiles(JobExecution jobExecution, FileDefinition fileExportDefinition, String tenantId) {
+    ExportedFile exportedFile = new ExportedFile()
+      .withFileId(fileExportDefinition.getId())
```
Comment (KVupp) [inline src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java]: we shouldn't be re-using the UUIDs, let;s have a unique Id for fileId

File: src/main/java/org/folio/dao/impl/JobExecutionDaoImpl.java [hunk 2]
```diff
@@ -49,13 +59,52 @@
...
+  public Future<JobExecution> update(JobExecution jobExecution, String tenantId) {
+    Promise<JobExecution> promise = Promise.promise();
+    try {
+      Criteria idCrit = constructCriteria(ID_FIELD, jobExecution.getId());
+      pgClientFactory.getInstance(tenantId).update(TABLE, jobExecution, new Criterion(idCrit), true, updateResult -> {
+        if (updateResult.failed()) {
+          LOGGER.error("Could not update jobExecution with id {}", jobExecution.getId(), updateResult.cause());
+          promise.fail(updateResult.cause());
+        } else if (updateResult.result().getUpdated() != 1) {
+          String errorMessage = String.format("JobExecution with id '%s' was not found", jobExecution.getId());
```
Comment (KVupp) [inline src/main/java/org/folio/dao/impl/JobExecutionDaoImpl.java]: Please change to use appropriate error message

File: src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java [hunk 3]
```diff
@@ -82,24 +87,69 @@ protected void initBlocking(JsonObject request, JsonObject params) {
...
+    jobExecutionService.getById(id, tenantId).compose(optionalJobExecution -> {
+      optionalJobExecution.map(jobExecution -> {
+        ExportedFile exportedFile = new ExportedFile()
+          .withFileId(UUID.randomUUID().toString())
+          .withFileName(fileExportDefinition.getFileName());
+        Set<ExportedFile> exportedFiles = jobExecution.getExportedFiles();
+        exportedFiles.add(exportedFile);
+        jobExecution.setExportedFiles(exportedFiles);
+        jobExecution.setStatus(JobExecution.Status.IN_PROGRESS);
+        jobExecution.setCompletedDate(new Date());
```
Comment (KVupp) [inline src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java]: why are we setting the completed date, when the job is still in progress?

File: src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java (hunk 1 above)
Comment (IlliaDaliek) [reply src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java]: Thank you for note, i will fix it

File: src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java (hunk 3 above)
Comment (IlliaDaliek) [reply src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java]: You are right, It is unnecessary here, will remove it

File: src/main/java/org/folio/dao/impl/JobExecutionDaoImpl.java [hunk 4]
```diff
@@ -58,27 +62,51 @@
...
+      Criteria idCrit = constructCriteria(ID_FIELD, jobExecutionId);
       pgClientFactory.getInstance(tenantId).get(TABLE, JobExecution.class, new Criterion(idCrit), false, promise);
     } catch (Exception e) {
       LOGGER.error(e);
       promise.fail(e);
     }
     return promise.future()
       .map(Results::getResults)
-      .map(jobExecutions -> jobExecutions.isEmpty() ? Optional.empty() : Optional.of(jobExecutions.get(0)));
+      .map(fileDefinitions -> fileDefinitions.isEmpty() ? Optional.empty() : Optional.of(fileDefinitions.get(0)));
```
Comment (KVupp) [inline src/main/java/org/folio/dao/impl/JobExecutionDaoImpl.java]: isn't this job executions?

Comment (IlliaDaliek) [reply src/main/java/org/folio/dao/impl/JobExecutionDaoImpl.java]: I've already corrected that, thank you

**Pull request merged**